augly/image/functional.py

```diff
@@ -521,6 +521,112 @@ def crop(
     return imutils.ret_and_save_image(aug_image, output_path)
 
 
+def distort_barrel(
+        image: Union[str, Image.Image],
+        output_path: Optional[str] = None,
+        a: float = 0.0,
+        b: float = 0.0,
+        c: float = 0.0,
+        d: float = 1.0,
+        metadata: Optional[List[Dict[str, Any]]] = None,
+):
+    """
+    Applies barrel distortion to the image with the following equation.
+
+        Rsrc = r * ( A*r3 + B*r2 + C*r + D )
+
+    @param image: the path to an image or a variable of type PIL.Image.Image
+        to be augmented
+
+    @param output_path: the path in which the resulting image will be stored.
+        If None, the resulting PIL Image will still be returned
+
+    @param a: Coefficient A in the equation Rsrc(r).
+
+    @param b: Coefficient B in the equation Rsrc(r).
+
+    @param c: Coefficient C in the equation Rsrc(r).
+
+    @param d: Coefficient D in the equation Rsrc(r).
```

Comment:
Can you give a bit of intuitive explanation of these params? e.g. "If you increase a, the barrel distortion becomes wider" or something like that. For now I'm unclear how changing each of these params might affect my images, so I would have to try applying the augmentation with different values for the params before decideing how to set them. It would be helpful for users to give them an idea right off the bat of what each param does. Also giving a suggested range for each would be great.

Reply:
Also, are there any invalid values of these params? Can I pass in `a=-1000`? If there is a valid range/set of values, please add asserts for each param to make sure they are in the valid range.

Reply:
1. I will add a bit more info to the parameter set, I initially thought that it'd better to give the users the link so they can inspect it visually. I provided that link on `BarrelDistortion` class, but not here.

2. There is no invalid values for those params.